$OpenBSD: patch-perdition_perdition_c,v 1.1 2013/11/15 10:10:02 sthen Exp $

Use the correct list of ciphers for outgoing STARTTLS cnnections.

CVE-2013-4584
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=729028

--- perdition/perdition.c.orig	Fri Nov 15 10:05:58 2013
+++ perdition/perdition.c	Fri Nov 15 10:06:31 2013
@@ -985,7 +985,7 @@ int main (int argc, char **argv, char **envp){
     else if((opt.ssl_mode & SSL_MODE_TLS_OUTGOING) &&
           (status & PROTOCOL_S_STARTTLS)) {
       server_io=perdition_ssl_client_connection(server_io, opt.ssl_ca_file, 
-		      opt.ssl_ca_path, opt.ssl_listen_ciphers, servername);
+		      opt.ssl_ca_path, opt.ssl_outgoing_ciphers, servername);
       if(!server_io) {
         VANESSA_LOGGER_DEBUG("perdition_ssl_connection outgoing");
         VANESSA_LOGGER_ERR("Fatal error establishing SSL connection");
